openpype/modules/sync_server/sync_server.py

```diff
@@ -229,6 +244,220 @@ def _get_configured_sites_from_setting(module, project_name, project_setting):
+        str: New local workfile path.
+    """
+
+    if not anatomy:
+        anatomy = Anatomy(project_name)
+
+    if not asset_doc:
+        asset_doc = get_asset_by_name(project_name, asset_name)
+
+    # Get sync server module
+    sync_server = ModulesManager().modules_by_name.get("sync_server")
+    if not sync_server or not sync_server.enabled:
+        print("Sync server module is disabled or unavailable.")
+        return
+
+    if subset_id is None:
+        print(
+            "Not any matched subset for task '{}' of '{}'.".format(
+                task_name, asset_name
+            )
+        )
+        return
+
+    if not workfile_representation:
+        print(
+            "Not published workfile for task '{}' and host '{}'.".format(
+                task_name, host_name
+            )
+        )
+        return
+
+    # If representation isn't available on remote site, then return.
+    if not sync_server.is_representation_on_site(
+        project_name,
+        workfile_representation["_id"],
+        sync_server.get_remote_site(project_name),
+    ):
+        print(
+            "Representation for task '{}' and host '{}'".format(
+                task_name, host_name
+            )
+        )
+        return
+
+    # Get local site
+    local_site_id = get_local_site_id()
+
+    # Add workfile representation to local site
+    representation_ids = {workfile_representation["_id"]}
+    representation_ids.update(
+        get_linked_representation_id(
+            project_name, repre_id=workfile_representation["_id"]
+        )
+    )
+    for repre_id in representation_ids:
+        sync_server.add_site(
+            project_name,
+            repre_id,
+            local_site_id,
+            force=True,
+            priority=99,
+            reset_timer=True,
+        )
+
+    # While representation unavailable locally, wait.
+    while not sync_server.is_representation_on_site(
+        project_name, workfile_representation["_id"], local_site_id
+    ):
+        sleep(5)
+
+    if not last_published_workfile_path:
+        last_published_workfile_path = get_last_published_workfile_path(
+            host_name,
+            project_name,
+            task_name,
+            workfile_representation,
+            anatomy=anatomy,
+        )
+
+    # Get workfile data
+    workfile_data = get_template_data_with_names(
+        project_name, asset_name, task_name, host_name
+    )
+
+    extension = last_published_workfile_path.split(".")[-1]
+
+    project_settings = get_project_settings(project_name)
+    template_key = get_workfile_template_key(
+        task_name, host_name, project_name, project_settings
+    )
+
+    # Get version patter for regex search
+    version_pattern = anatomy.templates[template_key]["version"]
+    version_pattern = re.sub(
+        r"{version.*?}",
+        r"([0-9]+)",
+        version_pattern,
+    )
+
+    # Get local workfile version number
+    last_local_workfile_version = None
```

Comment:
There is function `get_last_workfile_with_version` in `openpype.pipeline.workfile` which does exactly this.

Reply:
Changed with `get_last_workfile_with_version`